### DISCUSSION TITLE
Why abbreviation is not the default metric?

### DISCUSSION THREAD

**Discussion by barseghyanartur** [psincraian/pepy#459] [Ideas]
Stats of the [six](https://pepy.tech/project/six) look weird (as if it shows number of Gigabytes downloaded, not the total number of downloads). Why is abbreviation not the default metric?

**Comment by psincraian** [comment]
Because abbreviation is not a standard metric prefix https://en.m.wikipedia.org/wiki/Metric_prefix

**Comment by hugovk** [comment]
I'd argue that standard doesn't apply here and makes things more confusing (see https://github.com/psincraian/pepy/pull/180, https://github.com/psincraian/pepy/issues/192, https://github.com/psincraian/pepy/issues/346).

* We say a kilometre and gigabyte, but we don't say "a kilodownload" or "gigadownload".

* We don't use prefixes for "number of downloads".

* We say "a thousand downloads" or "a billion downloads".

"Number of downloads" are not an [SI unit: seconds, amperes, kelvin, etc](https://en.m.wikipedia.org/wiki/International_System_of_Units) nor used as a [non-metric units](https://en.m.wikipedia.org/wiki/Metric_prefix#Non-metric_units).

**Comment by psincraian** [comment]
To understand, are you suggesting to remove the prefixes? 🤔

**Comment by hugovk** [comment]
I'd suggest using B instead of G for billion, like in https://github.com/psincraian/pepy/pull/180.

**Comment by psincraian** [comment]
I know that some people complained, but there are only 4 complaints in 3 years I don't see is that bad as there is no standard. Some reasons I think G is better than B:

* People created ~10 times more personalized badges with SI units compared to abbreviation, so there is a clear preference.
* I know downloads is not an SI unit but I still think metric units prefixes are almost universal and every software engineer and datascientist is familiar with the term.
* Billion in some countries is 10E9 and in others is 10E12, so it can create confusion for other people.
* NPM downloads badge uses G instead of B for billion downloads.

**Comment by barseghyanartur** [comment]
@psincraian:

Honestly, I didn't know about the B vs G story until I recently bumped into the stats of the `six` package (one of the few that reached the billion score) which might be the answer to your remark about only 4 complaints in 3 years.

**Comment by hugovk** [comment]
Hi!

> * People created ~10 times more personalized badges with SI units compared to abbreviation, so there is a clear preference.

May be because so few are in the [billionaire's club](https://github.com/psincraian/pepy/pull/180#issue-352453418), that "System metric" is the default, and the badge creator doesn't explain the difference between "System metric" and "Abbreviation"

![image](https://user-images.githubusercontent.com/1324225/129538511-b2a0879f-ba76-4559-bb95-d82f9db088f2.png)


---

> * I know downloads is not an SI unit but I still think metric units prefixes are almost universal and every software engineer and datascientist is familiar with the term.

Sure, in the context of gigabytes, but not in the context of "number of things".

---

> * Billion in some countries is 10E9 and in others is 10E12, so it can create confusion for other people.

From https://github.com/psincraian/pepy/pull/180#issuecomment-565153122:

> The "million million" definition is historical, both American and British English now recognize 10^9 as a billion.

---

> * NPM downloads badge uses G instead of B for billion downloads.

Please could you point to one?

https://www.npmjs.com/package/lodash doesn't have a badge, but it does show:

> Weekly Downloads
> 39,341,043

![image](https://user-images.githubusercontent.com/1324225/129537635-f51d8f78-d00f-47ec-ad05-08ab636e4cef.png)

And doesn't have any prefix, SI or abbrev such as:

> Weekly Downloads
> 39M

Thanks!

**Comment by barseghyanartur** [comment]
@psincraian:

Hey, sorry, now I see your point. It's indeed very confusing (that various countries have different definitions for billion and onwards). I'm totally fine to leave it as is then.

**Comment by psincraian** [comment]
Hey,

> May be because so few are in the billionaire's club, that "System metric" is the default, and the badge creator doesn't explain the difference between "System metric" and "Abbreviation"

May be or may be not :-)

> Sure, in the context of gigabytes, but not in the context of "number of things".

Well, in all the International System of Units it applies (watt, ampere, metre, kilogram, etc). Also, as you point it's used for non-SI metrics also like bytes, but also decibels, and dollars.

From the definition from [Wikipedia](https://en.wikipedia.org/wiki/Unit_prefix)

_A unit prefix is a specifier or mnemonic that is prepended to units of measurement to indicate multiples or fractions of the units._

I think we can consider a download as a unit of measurement.

> The "million million" definition is historical, both American and British English now recognize 10^9 as a billion.

It's true for the USA and UK, but the other half of the world is using a billion as 10^9 https://en.wikipedia.org/wiki/Long_and_short_scale#Current_usage

> NPM downloads badge uses G instead of B for billion downloads.

Check lodash on shields.io: ![npm](https://img.shields.io/npm/dy/lodash) and here you can see the default [text formatter](https://github.com/badges/shields/blob/master/services/text-formatters.js#L51)

**Comment by psincraian** [comment]
I will convert this to a discussion :)
